Don't hide modal when `config.keyboard` is false

rohit2sharma95 opened the pull request:
Closes #30929

Modal will not be closed with <kbd>esc</kbd> key when modal's configuration has `keyboard:false` and backdrop is not static. It will trigger the `_triggerBackdropTransition()` method to show the animation though. But I am not sure if modal should show animation when pressing <kbd>esc</kbd> key and modal's backdrop is not static. 🤔 

Should I add/update the unit tests if it is okay? 🙂

rohit2sharma95 committed:
Don't hide modal when config.keyboard is false

Johann-S commented:
I don't know if I totally understood what you said, but the behavior should be: if a modal have `keyboard` at `false` in that case `esc` shouldn't work that's all

rohit2sharma95 commented:
> but the behavior should be: if a modal have `keyboard` at `false` in that case `esc` shouldn't work

Yes, <kbd>esc</kbd> is not working.
But modal is being animated even if its backdrop is not static. 🙂

XhmikosR committed:
Merge branch 'main' into modal-keyboard

rohit2sharma95 commented:
Should I add unit tests then, if it is fine 🙂

Johann-S commented:
yep please add a unit test 👌

rohit2sharma95 committed:
Update unit test

- Modal should not be closed when pressing esc key if keyboard = false
  and backdrop is 'static'

rohit2sharma95 commented:
@Johann-S Seems that updating the previous test would be okay here 🤔

Johann-S commented:
LGTM 👍 

/CC @XhmikosR

XhmikosR commented:
@Johann-S you forgot to approve?

XhmikosR commented on js/tests/unit/modal.spec.js:
```diff
@@ -595,12 +595,11 @@ describe('Modal', () => {
-    it('should not close modal when escape key is pressed with keyboard = false and backdrop = static', done => {
-      fixtureEl.innerHTML = '<div class="modal" data-bs-backdrop="static" data-bs-keyboard="false"><div class="modal-dialog"></div>'
+    it('should not close modal when escape key is pressed with keyboard = false', done => {
+      fixtureEl.innerHTML = '<div class="modal"><div class="modal-dialog"></div>'
```
Aren't we missing a `</div>` here BTW? It doesn't affect anything, and we don't enforce it, though.

rohit2sharma95 replied:
Updated @XhmikosR 👍

Johann-S commented:
I asked if you it was good to you too @XhmikosR 😉

XhmikosR commented:
Thank you both!

@rohit2sharma95 can you make a PR against v4-dev please?

rohit2sharma95 commented:
Okay sure @XhmikosR 👍 
Thanks 🙂
